src/utils/compoundApyHelpers.ts

```diff
@@ -29,6 +29,100 @@ export const tokenEarnedPerThousandDollarsCompounding = ({
   return parseFloat(interestEarned.toFixed(roundingDecimals))
 }
 
+/**
+ *
+ * @param investmentAmount - amount user wants to invest in USD
+ * @param apr - farm or pool apr as percentage. If its farm APR its only cake rewards APR without LP rewards APR
+ * @param earningTokenPrice - price of reward token
+ * @param compoundFrequency - how many compounds per 1 day, e.g. 1 = one per day, 0.142857142 - once per week
+ * @param performanceFee - performance fee as percentage
+ * @returns interest earned expressed in tokens
+ */
+export const getInterestBreakdown = ({
+  investmentAmount,
+  apr,
+  earningTokenPrice,
+  roundingDecimals = 2,
+  compoundFrequency = 1,
+  performanceFee = 0,
+}: {
+  investmentAmount: number
+  apr: number
+  earningTokenPrice: number
+  roundingDecimals?: number
+  compoundFrequency?: number
+  performanceFee?: number
+}) => {
+  // Everything here is worked out relative to a year, with the asset compounding at the compoundFrequency rate. 1 = once per day
+  const timesCompounded = 365 * compoundFrequency
+  // We use decimal values rather than % in the math for both APY and the number of days being calculates as a proportion of the year
+  let aprAsDecimal = apr / 100
+
+  if (performanceFee) {
+    // Reduce the APR by the % performance fee
+    const feeRelativeToApr = aprAsDecimal * performanceFee
+    const aprAfterFee = apr - feeRelativeToApr
+    aprAsDecimal = aprAfterFee / 100
+  }
+
+  return [1, 7, 30, 365].map((days) => {
```

Comment:
Not essential but we should consider making this a constant.

Reply:
Let's do this, these constants are reused everywhere, at least create an array

Reply:
fixed in 5d70e811b5b07ec28bca1ab35d3b9bebf8b5b149